api/jobs/handlers.py

```diff
@@ -482,6 +483,57 @@ def retry(self, _id):
         new_id = Queue.retry(j, force=True)
         return { "_id": new_id }
 
+
+    @require_drone
+    def prepare_complete(self, _id):
+        try:
+            j = Job.get(_id)
+        except Exception: # pylint: disable=broad-except
+            # TBD maybe raise APINotFound from Job.get?
```

Comment:
Yes I think that would be preferred. Core should hopefully only throw a generic `Exception` when it has no better error to throw, or perhaps via developer error (aka to alert a developer they misused a function).

Reply:
That should remove the redundant error checking below as well.

Reply:
Do we want to address @nagem's comment before merge?

Reply:
Retcon: I used github incorrectly and reviewed the wrong diff :+1: